portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppServiceImpl.java

```diff
@@ -150,6 +152,138 @@ public FileEntry addFileEntry(
 			byte[] bytes, ServiceContext serviceContext)
 		throws PortalException {
 
+		return addFileEntry(
+			null, repositoryId, folderId, sourceFileName, mimeType, title,
+			description, changeLog, bytes, serviceContext);
+	}
+
+	/**
+	 * Adds a file entry and associated metadata. It is created based on a
+	 * {@link File} object.
+	 *
+	 * <p>
+	 * This method takes two file names, the <code>sourceFileName</code> and the
+	 * <code>title</code>. The <code>sourceFileName</code> corresponds to the
+	 * name of the actual file being uploaded. The <code>title</code>
+	 * corresponds to a name the client wishes to assign this file after it has
+	 * been uploaded to the portal. If it is <code>null</code>, the <code>
+	 * sourceFileName</code> will be used.
+	 * </p>
+	 *
+	 * @param  repositoryId the primary key of the repository
+	 * @param  folderId the primary key of the file entry's parent folder
+	 * @param  sourceFileName the original file's name
+	 * @param  mimeType the file's MIME type
+	 * @param  title the name to be assigned to the file (optionally <code>null
+	 *         </code>)
+	 * @param  description the file's description
+	 * @param  changeLog the file's version change log
+	 * @param  file the file's data (optionally <code>null</code>)
+	 * @param  serviceContext the service context to be applied. Can set the
+	 *         asset category IDs, asset tag names, and expando bridge
+	 *         attributes for the file entry. In a Liferay repository, it may
+	 *         include:  <ul> <li> fileEntryTypeId - ID for a custom file entry
+	 *         type </li> <li> fieldsMap - mapping for fields associated with a
+	 *         custom file entry type </li> </ul>
+	 * @return the file entry
+	 * @throws PortalException if a portal exception occurred
```

Comment:
I think you have to add an @review

Reply:
I've made a quick search and the first review javadoc annotation that I've found is from August 2019. Let's hope they review these comments a little bit quicker 😄

Reply:
😂 I think documentation is swamped with liferay.learn and is ignoring those